src/App.tsx

```diff
@@ -1,14 +1,32 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState(true);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    getTodos()
+      .then(serverTodos => {
+        setTodos(serverTodos);
+        setFilteredTodos(serverTodos);
+        setIsLoading(false);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+  }, []);
```

Comment:
1. I don't agree that we need to have a separate state for `filteredTodos`, let's just filter `todos` on each render. (Basically make `filteredTodos` a regular variable)
2. Let's use `finally` and put it there `setIsLoading(false);`